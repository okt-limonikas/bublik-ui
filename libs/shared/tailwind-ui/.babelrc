{
	"presets": [
		[
			"@nx/react/babel",
			{
				"runtime": "automatic",
				"useBuiltIns": "usage"
			}
		],
		"@emotion/babel-preset-css-prop"
	],
	"plugins": []
}
