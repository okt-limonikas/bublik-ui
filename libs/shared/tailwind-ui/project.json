{
	"name": "shared-tailwind-ui",
	"$schema": "../../../node_modules/nx/schemas/project-schema.json",
	"sourceRoot": "libs/shared/tailwind-ui/src",
	"projectType": "library",
	"targets": {
		"lint": {
			"executor": "@nx/eslint:lint",
			"options": {
				"lintFilePatterns": ["libs/shared/tailwind-ui/**/*.{ts,tsx,js,jsx}"]
			}
		},
		"test": {
			"executor": "@nx/vite:test",
			"outputs": ["{workspaceRoot}/coverage/libs/shared/tailwind-ui"],
			"options": {
				"passWithNoTests": true
			}
		}
	},
	"tags": []
}
