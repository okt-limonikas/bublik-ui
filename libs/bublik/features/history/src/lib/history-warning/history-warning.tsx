/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { ReactNode } from 'react';

import { ButtonTw, Icon } from '@/shared/tailwind-ui';

export interface HistoryWarningProps {
	icon: ReactNode;
	label: string;
	description: string;
	onButtonClick?: () => void;
}

export const HistoryWarning = (props: HistoryWarningProps) => {
	return (
		<div className="grid place-items-center h-[calc(100vh-256px)]">
			<div className="flex flex-col items-center text-center">
				{props.icon}
				<h3 className="mt-2 text-sm font-medium text-gray-900">
					{props.label}
				</h3>
				<p className="mt-1 text-sm text-gray-500">{props.description}</p>
				{props.onButtonClick ? (
					<div className="mt-6">
						<ButtonTw
							type="button"
							variant="primary"
							rounded="lg"
							size="xs"
							onClick={props.onButtonClick}
						>
							<Icon name="Filter" className="mr-2" />
							Open Search
						</ButtonTw>
					</div>
				) : null}
			</div>
		</div>
	);
};
