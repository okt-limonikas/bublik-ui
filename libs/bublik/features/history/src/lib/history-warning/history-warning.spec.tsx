/* SPDX-License-Identifier: Apache-2.0 */
/* SPDX-FileCopyrightText: 2021-2023 OKTET Labs Ltd. */
import { render } from '@testing-library/react';

import { Icon } from '@/shared/tailwind-ui';

import { HistoryWarning } from '../history-warning';

describe('HistoryWarning component', () => {
	it('should render correctly', () => {
		const { asFragment } = render(
			<HistoryWarning
				icon={<Icon name="Filter" />}
				label="label"
				description="description"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
