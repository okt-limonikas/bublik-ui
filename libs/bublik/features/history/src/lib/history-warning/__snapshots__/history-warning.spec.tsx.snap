// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HistoryWarning component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="grid place-items-center h-[calc(100vh-256px)]"
  >
    <div
      class="flex flex-col items-center text-center"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.73 17.823c0 .442.346.803.772.803.426 0 .772-.36.772-.803v-3.984a.82.82 0 0 0-.21-.55L3.703 6.216a.623.623 0 0 1-.158-.416V4.217c0-.335.263-.608.585-.608h3.089c.426 0 .772-.36.772-.804C7.99 2.36 7.644 2 7.218 2H4.129C2.955 2 2 2.994 2 4.217v1.582c0 .566.204 1.103.575 1.516l6.155 6.842v3.666Zm.16 3.718c.13.289.41.459.7.459.11 0 .22-.024.325-.074l4.321-2.07c.271-.13.446-.41.446-.72v-5.09l6.679-6.756c.406-.41.639-.98.639-1.565V4.193C22 2.983 21.044 2 19.87 2h-8.899a.785.785 0 0 0-.772.795c0 .44.346.797.772.797h8.899c.323 0 .585.27.585.601v1.532a.61.61 0 0 1-.175.428l-6.91 6.993a.808.808 0 0 0-.233.567v4.915l-3.873 1.855a.807.807 0 0 0-.375 1.058Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <h3
        class="mt-2 text-sm font-medium text-gray-900"
      >
        label
      </h3>
      <p
        class="mt-1 text-sm text-gray-500"
      >
        description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
